test/EFCore.MySql.FunctionalTests/Scaffolding/MySqlDatabaseModelFactoryTest.cs

```diff
@@ -731,6 +731,42 @@ public void Set_referential_action_for_foreign_key()
 DROP TABLE PrincipalTable;");
         }
 
+        // Important: the inconsistent casing of the referenced table name is intentional. Do not change.
+        [Fact]
+        public void Ensure_constraints_scaffold_with_case_mismatch()
+        {
+            Test(
+                @"
+CREATE TABLE `PrincipalTable` (
+  `Id` INT NOT NULL,
+  PRIMARY KEY (`Id`));
+
+CREATE TABLE `DependentTable` (
+  `Id` INT NOT NULL,
+  `ForeignKeyId` INT NOT NULL,
+  PRIMARY KEY (`Id`),
+  CONSTRAINT `ForeignKey_Id`
+    FOREIGN KEY (`ForeignKeyId`)
+    REFERENCES `principaltable` (`Id`)
+);",
+                Enumerable.Empty<string>(),
+                Enumerable.Empty<string>(),
+                dbModel =>
+                {
+                    var principal = dbModel.Tables.FirstOrDefault(t => t.Name == "PrincipalTable");
+                    var dependent = dbModel.Tables.FirstOrDefault(t => t.Name == "DependentTable");
+
+                    Assert.NotNull(principal);
+                    Assert.NotNull(dependent);
+
+                    Assert.Contains(dependent.ForeignKeys, t => t.PrincipalTable.Name == principal.Name);
+                    Assert.True(true);
```

Comment:
Just remove this line and move the comment from outside the method inside the method, so it does not get lost later on. We can merge then.

Reply:
Done